### PR TITLE
Support long contexts with LoRA

### PR DESCRIPTION
**Pull request by SanjuCSudhakaran**
This PR enables long-contexts support with LoRA

### CODE REVIEW

File: vllm/lora/punica.py [hunk 1]
```diff
@@ -119,11 +116,13 @@ def convert_mapping(
         embedding_indices[i] = lora_idx if index_mapping_indices[i] > 0 else 0
         lora_indices[i] = lora_idx
         if long_lora_context:
-            assert long_lora_offsets is not None
             lora_offset: int = long_lora_context.offsets_by_lora_id.get(
                 index_mapping_indices[i], 0)
-            long_lora_offsets[i] = lora_offset
+            long_lora_offsets_list.append(lora_offset)
 
+    long_lora_offsets = torch.tensor(long_lora_offsets_list,
```
Comment (vivekgoe) [inline vllm/lora/punica.py:123]: W.r.t. functionality this change is ok for other backends (e.g. CUDA) also. @kzawora-intel  is it ok to keep it? or do we need to make it HPU specific to be more upstream(ing) friendly?

File: vllm/worker/hpu_model_runner.py [hunk 2]
```diff
@@ -649,12 +650,30 @@ def load_model(self) -> None:
                 assert hasattr(
                     self.model, "embedding_padding_modules"
                 ), "Model does not have embedding_padding_modules"
+
+                if supports_multimodal(self.model):
```
Comment (vivekgoe) [inline vllm/worker/hpu_model_runner.py:654]: Is this code coming from rebased version of "model_runner.py"? or is this something we have added on our own?

File: vllm/lora/punica.py (hunk 1 above)
Comment (michalkuligowski) [reply vllm/lora/punica.py:123]: Yes we need to keep code upstreamable, please add check for hpu

**Review by michalkuligowski — CHANGES_REQUESTED** [review]

File: vllm/lora/punica.py (hunk 1 above)
Comment (SanjuCSudhakaran) [reply vllm/lora/punica.py]: Done

**Pull request merged**